src/reducers.js

```diff
@@ -70,38 +79,22 @@ function organisations(
     case REQUEST_ORGANISATIONS:
       return { ...state, isFetching: true };
     case RECEIVE_ORGANISATIONS:
+    {
       return {
         ...state,
-        available: action.data
-          .filter(e => {
-            return (
-              e.roles.find(e => e === "manager") ||
-              e.roles.find(e => e === "admin") ||
-              e.roles.find(e => e === "supplier") ||
-              e.roles.find(e => e === "user")
-            );
-          })
-          .map(organisation => {
-            //use organisation uuid without dashes only
-            return {
-              ...organisation,
-              uuid: organisation.uuid.replace(/-/g, "")
-            };
-          }),
-        availableForRasterSharedWith: action.data.map(organisation => {
-          //use organisation uuid without dashes only
-          return {
-            ...organisation,
-            uuid: organisation.uuid.replace(/-/g, "")
-          };
-        }),
+        available: action.available,
+        availableForRasterSharedWith: action.all,
         isFetching: false,
         timesFetched: state.timesFetched + 1
       };
+    }
     case SELECT_ORGANISATION:
       // the api v3 accepts no dashes in the uuid (this is called unique_id in api v3)
       const selectedOrganisation = action.organisation;
-      selectedOrganisation.uuid = selectedOrganisation.uuid.replace(/-/g, "");
+      if (selectedOrganisation) {
+        selectedOrganisation.uuid = selectedOrganisation.uuid.replace(/-/g, "");
+      }
```

Comment:
Shame

Reply:
True, but this problem is not introduced in this pr.
Last attempt to change this created a bug on production.
Can I fix this in a new pr?
https://nelen-schuurmans.atlassian.net/browse/FRNT-555